src/Menu/Menu.spec.js

```diff
@@ -1,15 +1,101 @@
 /* eslint-env mocha */
 import React from 'react';
-import {shallow} from 'enzyme';
+import {mount, shallow} from 'enzyme';
+import {match, spy} from 'sinon';
 import {assert} from 'chai';
 import Menu from './Menu';
 import MenuItem from '../MenuItem';
 import Divider from '../Divider';
 import getMuiTheme from '../styles/getMuiTheme';
+import keycode from 'keycode';
 
 describe('<Menu />', () => {
   const muiTheme = getMuiTheme();
   const shallowWithContext = (node) => shallow(node, {context: {muiTheme}});
+  const mountWithContext = (node) => mount(node, {context: {muiTheme}});
+  const keycodeEvent = (key) => ({keyCode: keycode(key)});
+
+  describe('onMenuItemFocusChange', () => {
+    function createMenu(props) {
+      return (
+        <Menu {...props}>
+          <MenuItem primaryText="item 1" />
+          <Divider />
+          <MenuItem primaryText="item 2" />
+          <MenuItem primaryText="item 3" />
+        </Menu>
+      );
+    }
+
+    it('is invoked when using the arrow key to go down to the bottom and back up to the top', () => {
+      const onMenuItemFocusChangeSpy = spy();
+      const menu = createMenu({
+        disableAutoFocus: false,
+        onMenuItemFocusChange: onMenuItemFocusChangeSpy,
+      });
+      const wrapper = mountWithContext(menu);
+
+      assert(onMenuItemFocusChangeSpy.calledWith(null, 0),
+             'initial focus should invoke callback with 0');
+      onMenuItemFocusChangeSpy.reset();
```

Comment:
I am very interested in making sure we don't break what happens when the user enters a particular *sequence* of arrow key events.  That would be difficult for me to test if I isolated each arrow key event inside its own `it` since I would lose all the `Menu` state along with the spy state.

Does that make sense?  I'm not confident I interpreted your comment correctly.